src/Validation/Rules/HasRectorRule.php

```diff
@@ -7,64 +7,58 @@
 use Closure;
 use Illuminate\Contracts\Validation\ValidationRule;
 use PhpParser\Error;
-use PhpParser\Node;
-use PhpParser\Node\Expr\MethodCall;
-use PhpParser\Node\Identifier;
-use PhpParser\NodeFinder;
-use PhpParser\ParserFactory;
+use Rector\Config\RectorConfig;
+use Rector\Contract\Rector\RectorInterface;
+use Rector\DependencyInjection\LazyContainerFactory;
 use Rector\Rector\AbstractRector;
+use Symfony\Component\Filesystem\Filesystem;
 
+/**
+ * @see \App\Tests\Validator\Rules\HasRectorRule\HasRectorRuleTest
+ */
 final class HasRectorRule implements ValidationRule
 {
     public function validate(string $attribute, mixed $value, Closure $fail): void
     {
-
         // dummy check for custom rule request
         if (str_contains($value, AbstractRector::class)) {
             return;
         }
 
-        // @todo load the config and simply see if any rule is loaded
-        $parserFactory = new ParserFactory();
-        $parser = $parserFactory->create(ParserFactory::PREFER_PHP7);
-
         try {
-            $stmts = $parser->parse($value);
-            $nodeFinder = new NodeFinder();
+            $filesystem = new Filesystem();
+
+            $configFilePath = sys_get_temp_dir() . '/temp-rector-config.php';
+            $filesystem->dumpFile($configFilePath, $value);
```

Comment:
This might be problematic spot on production

Reply:
Just deployed and looks good:

![Screenshot from 2024-07-11 23-57-34](https://github.com/user-attachments/assets/c198480a-52da-4fa8-8d36-7afb385d055d)

Reply:
this seems cause possible error https://github.com/rectorphp/getrector-com/issues/2438, also for consecutive request, the file match replace each other, so it may need random name, and remove after it usage.

Reply:
https://github.com/rectorphp/getrector-com/pull/2446